atlasclient/utils.py

```diff
@@ -21,12 +21,20 @@
 
 class JSONObject(dict):
     """Dictionary with dot-notation read access."""
+    def __coerce_dict(self, value):
+        if isinstance(value, dict):
+            return type(self)(value)
+        return value
+
     def __getattr__(self, name):
         if name in self:
-            return self[name]
-        raise AttributeError('{} has no property named {}}.'.format(
+            return self.__coerce_dict(self[name])
+        raise AttributeError('{} has no property named {}.'.format(
             self.__class__.__name__, name))
 
+    def __getitem__(self, item):
+        return self.__coerce_dict(super().__getitem__(item))
+
```

Comment:
Why do we need these changes?

Reply:
This allows dot-access to nested dictionary members which is used quite often. With the older code only the first-level keys of the dictionary could be accessed via the `obj.attr` pattern. This change ensures that this kind of access is possible for all nested dicts as well.